CI looks like broken

https://github.com/apache/iceberg-rust/actions/runs/8323042113/job/22771908103?pr=258

```
warning: use of deprecated function `arrow_arith::temporal::month_dyn`: Use `date_part` instead
  --> crates/iceberg/src/transform/temporal.rs:22:16
   |
22 |     temporal::{month_dyn, year_dyn},
   |                ^^^^^^^^^
   |
   = note: `#[warn(deprecated)]` on by default

warning: use of deprecated function `arrow_arith::temporal::year_dyn`: Use `date_part` instead
  --> crates/iceberg/src/transform/temporal.rs:22:27
   |
22 |     temporal::{month_dyn, year_dyn},
   |                           ^^^^^^^^

warning: use of deprecated function `arrow_arith::temporal::year_dyn`: Use `date_part` instead
  --> crates/iceberg/src/transform/temporal.rs:47:13
   |
47 |             year_dyn(&input).map_err(|err| Error::new(ErrorKind::Unexpected, format!("{err}")))?;
   |             ^^^^^^^^

warning: use of deprecated function `arrow_arith::temporal::year_dyn`: Use `date_part` instead
  --> crates/iceberg/src/transform/temporal.rs:65:13
   |
65 |             year_dyn(&input).map_err(|err| Error::new(ErrorKind::Unexpected, format!("{err}")))?;
   |             ^^^^^^^^

warning: use of deprecated function `arrow_arith::temporal::month_dyn`: Use `date_part` instead
  --> crates/iceberg/src/transform/temporal.rs:72:13
   |
72 |             month_dyn(&input).map_err(|err| Error::new(ErrorKind::Unexpected, format!("{err}")))?;
   |             ^^^^^^^^^

error[E0308]: mismatched types
   --> crates/iceberg/src/writer/file_writer/parquet_writer.rs:118:13
    |
116 |         let writer = AsyncArrowWriter::try_new(
    |                      ------------------------- arguments to this function are incorrect
117 |             inner_writer,
118 |             self.schema.clone(),
    |             ^^^^^^^^^^^^^^^^^^^ expected `arrow_schema::schema::Schema`, found `arrow_schema::Schema`
    |
    = note: `arrow_schema::Schema` and `arrow_schema::schema::Schema` have similar names, but are actually distinct types
note: `arrow_schema::Schema` is defined in crate `arrow_schema`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-schema-51.0.0/src/schema.rs:187:1
    |
187 | pub struct Schema {
    | ^^^^^^^^^^^^^^^^^
note: `arrow_schema::schema::Schema` is defined in crate `arrow_schema`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-schema-50.0.0/src/schema.rs:181:1
    |
181 | pub struct Schema {
    | ^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `arrow_schema` are being used?
note: associated function defined here
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/parquet-50.0.0/src/arrow/async_writer/mod.rs:95:12
    |
95  |     pub fn try_new(
    |            ^^^^^^^

error[E0308]: mismatched types
   --> crates/iceberg/src/writer/file_writer/parquet_writer.rs:220:27
    |
220 |         self.writer.write(batch).await.map_err(|err| {
    |                     ----- ^^^^^ expected `RecordBatch`, found a different `RecordBatch`
    |                     |
    |                     arguments to this method are incorrect
    |
    = note: `RecordBatch` and `RecordBatch` have similar names, but are actually distinct types
note: `RecordBatch` is defined in crate `arrow_array`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-array-51.0.0/src/record_batch.rs:72:1
    |
72  | pub struct RecordBatch {
    | ^^^^^^^^^^^^^^^^^^^^^^
note: `RecordBatch` is defined in crate `arrow_array`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-array-50.0.0/src/record_batch.rs:72:1
    |
72  | pub struct RecordBatch {
    | ^^^^^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `arrow_array` are being used?
note: method defined here
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/parquet-50.0.0/src/arrow/async_writer/mod.rs:116:18
    |
116 |     pub async fn write(&mut self, batch: &RecordBatch) -> Result<()> {
    |                  ^^^^^

```

Thanks for reporting this @viirya. I'm seeing something similar on my local machine:

```
error[E0308]: mismatched types
   --> /Users/fokkodriesprong/.cargo/git/checkouts/iceberg-rust-d49e83c40ef4cf40/d6703df/crates/iceberg/src/writer/file_writer/parquet_writer.rs:118:13
    |
116 |         let writer = AsyncArrowWriter::try_new(
    |                      ------------------------- arguments to this function are incorrect
117 |             inner_writer,
118 |             self.schema.clone(),
    |             ^^^^^^^^^^^^^^^^^^^ expected `arrow_schema::schema::Schema`, found `arrow_schema::Schema`
    |
    = note: `arrow_schema::Schema` and `arrow_schema::schema::Schema` have similar names, but are actually distinct types
note: `arrow_schema::Schema` is defined in crate `arrow_schema`
   --> /Users/fokkodriesprong/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-schema-51.0.0/src/schema.rs:187:1
    |
187 | pub struct Schema {
    | ^^^^^^^^^^^^^^^^^
note: `arrow_schema::schema::Schema` is defined in crate `arrow_schema`
   --> /Users/fokkodriesprong/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-schema-50.0.0/src/schema.rs:181:1
    |
181 | pub struct Schema {
    | ^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `arrow_schema` are being used?
note: associated function defined here
   --> /Users/fokkodriesprong/.cargo/registry/src/index.crates.io-6f17d22bba15001f/parquet-50.0.0/src/arrow/async_writer/mod.rs:95:12
    |
95  |     pub fn try_new(
    |            ^^^^^^^
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -18,10 +18,10 @@
 [workspace]
 resolver = "2"
 members = [
-  "crates/catalog/*",
-  "crates/examples",
-  "crates/iceberg",
-  "crates/test_utils",
+    "crates/catalog/*",
+    "crates/examples",
+    "crates/iceberg",
+    "crates/test_utils",
 ]
 
 [workspace.package]
@@ -37,9 +37,9 @@ rust-version = "1.75.0"
 anyhow = "1.0.72"
 apache-avro = "0.16"
 array-init = "2"
-arrow-arith = { version = ">=46" }
-arrow-array = { version = ">=46" }
-arrow-schema = { version = ">=46" }
+arrow-arith = { version = "51" }
+arrow-array = { version = "51" }
+arrow-schema = { version = "51" }
 async-stream = "0.3.5"
 async-trait = "0.1"
 bimap = "0.6"
@@ -61,7 +61,7 @@ murmur3 = "0.5.2"
 once_cell = "1"
 opendal = "0.45"
 ordered-float = "4.0.0"
-parquet = "50"
+parquet = "51"
 pilota = "0.10.0"
 pretty_assertions = "1.4.0"
 port_scanner = "0.1.5"

--- a/crates/iceberg/src/transform/temporal.rs
+++ b/crates/iceberg/src/transform/temporal.rs
@@ -17,10 +17,8 @@
 
 use super::TransformFunction;
 use crate::{Error, ErrorKind, Result};
-use arrow_arith::{
-    arity::binary,
-    temporal::{month_dyn, year_dyn},
-};
+use arrow_arith::temporal::DatePart;
+use arrow_arith::{arity::binary, temporal::date_part};
 use arrow_array::{
     types::Date32Type, Array, ArrayRef, Date32Array, Int32Array, TimestampMicrosecondArray,
 };
@@ -43,8 +41,8 @@ pub struct Year;
 
 impl TransformFunction for Year {
     fn transform(&self, input: ArrayRef) -> Result<ArrayRef> {
-        let array =
-            year_dyn(&input).map_err(|err| Error::new(ErrorKind::Unexpected, format!("{err}")))?;
+        let array = date_part(&input, DatePart::Year)
+            .map_err(|err| Error::new(ErrorKind::Unexpected, format!("{err}")))?;
         Ok(Arc::<Int32Array>::new(
             array
                 .as_any()
@@ -61,15 +59,15 @@ pub struct Month;
 
 impl TransformFunction for Month {
     fn transform(&self, input: ArrayRef) -> Result<ArrayRef> {
-        let year_array =
-            year_dyn(&input).map_err(|err| Error::new(ErrorKind::Unexpected, format!("{err}")))?;
+        let year_array = date_part(&input, DatePart::Year)
+            .map_err(|err| Error::new(ErrorKind::Unexpected, format!("{err}")))?;
         let year_array: Int32Array = year_array
             .as_any()
             .downcast_ref::<Int32Array>()
             .unwrap()
             .unary(|v| 12 * (v - UNIX_EPOCH_YEAR));
-        let month_array =
-            month_dyn(&input).map_err(|err| Error::new(ErrorKind::Unexpected, format!("{err}")))?;
+        let month_array = date_part(&input, DatePart::Month)
+            .map_err(|err| Error::new(ErrorKind::Unexpected, format!("{err}")))?;
         Ok(Arc::<Int32Array>::new(
             binary(
                 month_array.as_any().downcast_ref::<Int32Array>().unwrap(),

--- a/crates/iceberg/src/writer/file_writer/parquet_writer.rs
+++ b/crates/iceberg/src/writer/file_writer/parquet_writer.rs
@@ -18,7 +18,6 @@
 //! The module contains the file writer for parquet file format.
 
 use std::{
-    cmp::max,
     collections::HashMap,
     sync::{atomic::AtomicI64, Arc},
 };
@@ -43,9 +42,6 @@ use super::{
 /// ParquetWriterBuilder is used to builder a [`ParquetWriter`]
 #[derive(Clone)]
 pub struct ParquetWriterBuilder<T: LocationGenerator, F: FileNameGenerator> {
-    /// `buffer_size` determines the initial size of the intermediate buffer.
-    /// The intermediate buffer will automatically be resized if necessary
-    init_buffer_size: usize,
     props: WriterProperties,
     schema: ArrowSchemaRef,
 
@@ -55,13 +51,9 @@ pub struct ParquetWriterBuilder<T: LocationGenerator, F: FileNameGenerator> {
 }
 
 impl<T: LocationGenerator, F: FileNameGenerator> ParquetWriterBuilder<T, F> {
-    /// To avoid EntiryTooSmall error, we set the minimum buffer size to 8MB if the given buffer size is smaller than it.
-    const MIN_BUFFER_SIZE: usize = 8 * 1024 * 1024;
-
     /// Create a new `ParquetWriterBuilder`
     /// To construct the write result, the schema should contain the `PARQUET_FIELD_ID_META_KEY` metadata for each field.
     pub fn new(
-        init_buffer_size: usize,
         props: WriterProperties,
         schema: ArrowSchemaRef,
         file_io: FileIO,
@@ -69,7 +61,6 @@ impl<T: LocationGenerator, F: FileNameGenerator> ParquetWriterBuilder<T, F> {
         file_name_generator: F,
     ) -> Self {
         Self {
-            init_buffer_size,
             props,
             schema,
             file_io,
@@ -112,20 +103,14 @@ impl<T: LocationGenerator, F: FileNameGenerator> FileWriterBuilder for ParquetWr
                 .generate_location(&self.file_name_generator.generate_file_name()),
         )?;
         let inner_writer = TrackWriter::new(out_file.writer().await?, written_size.clone());
-        let init_buffer_size = max(Self::MIN_BUFFER_SIZE, self.init_buffer_size);
-        let writer = AsyncArrowWriter::try_new(
-            inner_writer,
-            self.schema.clone(),
-            init_buffer_size,
-            Some(self.props),
-        )
-        .map_err(|err| {
-            Error::new(
-                crate::ErrorKind::Unexpected,
-                "Failed to build parquet writer.",
-            )
-            .with_source(err)
-        })?;
+        let writer = AsyncArrowWriter::try_new(inner_writer, self.schema.clone(), Some(self.props))
+            .map_err(|err| {
+                Error::new(
+                    crate::ErrorKind::Unexpected,
+                    "Failed to build parquet writer.",
+                )
+                .with_source(err)
+            })?;
 
         Ok(ParquetWriter {
             writer,
@@ -311,7 +296,6 @@ mod tests {
 
         // write data
         let mut pw = ParquetWriterBuilder::new(
-            0,
             WriterProperties::builder().build(),
             to_write.schema(),
             file_io.clone(),
@@ -551,7 +535,6 @@ mod tests {
 
         // write data
         let mut pw = ParquetWriterBuilder::new(
-            0,
             WriterProperties::builder().build(),
             to_write.schema(),
             file_io.clone(),

